blaze/expr/core.py

```diff
@@ -404,6 +404,7 @@ def subterms(x):
     yield x
 
 
+@memoize_mutable
 def subs(o, d):
```

Comment:
@llllllllll to your knowledge in Blaze, is `subs()` ever used for anything besides expressions?  IOW, even though `subs` is fairly general and can handle mutable nested data structures, in actuality is it ever used for (mutable) data structures that aren't expressions?


I ask because having mutable memoization starts adding complexity that we may not need if we constrain `subs` to immutable things.